netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -79,14 +79,18 @@
    * noticing changes to DNS.
    */
   public static NettyChannelBuilder forAddress(SocketAddress serverAddress) {
```

Comment:
The fact that the `forTarget` method was recently added speaks to the fact that `forAddress` in its current form is broken. Clients that were already written against `forAddress` previously did not see our new nifty disclaimer about its usage.  They may not see a problem yet, but they might in the future and it won't be clear what the problem is.  Fixing `forAddress` to do the *right* thing is a much safer approach WRT users.  Preventing users from shooting themselves in the foot should be a goal in general.